internal/service/advancedclustertpf/resource_compatibility_reuse.go

```diff
@@ -17,6 +18,54 @@ func FormatMongoDBMajorVersion(version string) string {
 	return fmt.Sprintf("%.1f", cast.ToFloat32(version))
 }
 
+func resolveExtraAPIInfo(ctx context.Context, projectID string, cluster *admin.ClusterDescription20240805, api admin.NetworkPeeringApi) (*ExtraAPIInfo, error) {
+	containerIDs := map[string]string{}
+	for _, spec := range cluster.GetReplicationSpecs() {
+		for _, regionConfig := range spec.GetRegionConfigs() {
+			providerName := regionConfig.GetProviderName()
+			if providerName == constant.TENANT {
+				continue
+			}
+			params := &admin.ListPeeringContainerByCloudProviderApiParams{
+				GroupId:      projectID,
+				ProviderName: &providerName,
+			}
+			containerIDKey := fmt.Sprintf("%s:%s", providerName, regionConfig.GetRegionName())
+			if _, ok := containerIDs[containerIDKey]; ok {
+				continue
+			}
+			containers, _, err := api.ListPeeringContainerByCloudProviderWithParams(ctx, params).Execute()
+			if err != nil {
+				return nil, err
+			}
+			if results := getAdvancedClusterContainerID(containers.GetResults(), &regionConfig); results != "" {
```

Comment:
empty results is not an error?

Reply:
Re-used logic from before, but I agree, better to raise an error

Reply:
changed in f29a7510cb2bd5c2774ca346d0ac7d858a3baabd